pandas/core/groupby/ops.py

```diff
@@ -655,15 +655,15 @@ def agg_series(self, obj, func):
             return self._aggregate_series_fast(obj, func)
         except AssertionError:
             raise
-        except ValueError as err:
+        except (ValueError, AttributeError, TypeError) as err:
             if "No result." in str(err):
                 # raised in libreduction
                 pass
             elif "Function does not reduce" in str(err):
                 # raised in libreduction
                 pass
             else:
-                raise
+                pass
```

Comment:
@jbrockmendel the whole block is now basically an `except (..): pass` (so I could make it much shorter), but you might have put in those specific checks as pointers to what to clean up later?

Reply:
so something is being done with a DecimalArray that raises a TypeError?  and i guess by not-catching it here it is getting caught somewhere above that isn't doing re-casting appropriately?  can we track down where that is?

Reply:
It looks like in _aggregate_series_fast there is a call to libreduction that tries to assign this DecimalArray to a name that libreduction has typed as an ndarray, which raises TypeError

Reply:
But that is expected I think?

Reply:
we get to decide what is "expected"; maybe i dont understand what you're getting at

Reply:
Well, we try the fast path, and if that fails, we go for the slow path. So here it fails, so we choose the slow path. That is how I understand that it is currently designed.

Reply:
so if we fix the issue in libreduction, there's a chance that the fast path will work

Reply:
libreduction is not meant to work for all cases, eg if the values is not an ndarray. Yes, we could handle that differently (eg check in advance if it should work), but that is not the approach that is currently taken. I would prefer to limit the scope of this PR to adding the test cases and restoring the old behaviour with a minimal change.

Reply:
OK, how about add the test and xfail it.  i'll get to it shortly

Reply:
If you'll get to it shortly, how about merging this (with further edits based on feedback of course), and at least fixing the regression on master?

Reply:
To clarify: I would personally prefer to merge this without xfail to have the coverage on master. You're doing other PRs cleaning things up in groupby, and it would be good to ensure this keeps working

Reply:
fair enough

Reply:
actually, can you do the same `str(err)` checking so that we only let through the relevant TypeErrors?  One of the other PRs afoot is specifically targeting other TypeErrors

Reply:
That's an error message coming from another library though (cython?). Do we have guarantee that that is stable? 
(the other two errors that are catched that way are raised by our own code in reduction.pyx)

Reply:
> Do we have guarantee that that is stable?

If nothing else, you can check that both "ndarray" and "DecimalArray" are present.  I'm sure you can figure something out.

Reply:
This is not only for DecimalArray, but for any kind of internal/external EA.

Reply:
But can add the check for just ndarray.